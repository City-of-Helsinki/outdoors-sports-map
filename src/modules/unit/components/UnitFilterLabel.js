import React from 'react';
import { withTranslation } from 'react-i18next';

// eslint-disable-next-line react/prop-types
const UnitFilterLabel = ({ message, id }) => {
  if (!message) {
    return null;
  }
  return (
    <div className="unit-filter-label" id={id}>
      <span>{message}:</span>
    </div>
  );
};

export default withTranslation()(UnitFilterLabel);
