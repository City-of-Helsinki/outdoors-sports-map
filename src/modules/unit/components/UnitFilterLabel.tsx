import PropTypes from "prop-types";
import React from "react";

function UnitFilterLabel({ message, id }) {
  if (!message) {
    return null;
  }

  return (
    <div className="unit-filter-label" id={id}>
      <span>{message}:</span>
    </div>
  );
}

UnitFilterLabel.defaultProps = {
  id: undefined,
};
UnitFilterLabel.propTypes = {
  message: PropTypes.string.isRequired,
  id: PropTypes.string,
};

export default UnitFilterLabel;
