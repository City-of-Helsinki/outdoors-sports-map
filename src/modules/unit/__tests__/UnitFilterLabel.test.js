import React from 'react';
import { shallow } from 'enzyme';

import UnitFilterLabel from '../components/UnitFilterLabel';

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  translate: () => (Component) => {
    // eslint-disable-next-line no-param-reassign
    Component.defaultProps = { ...Component.defaultProps, t: (k) => k };
    return Component;
  },
}));

it('renders snapshot correctly', () => {
  const unitFilterLabel = shallow(<UnitFilterLabel filterName="sport" />);
  expect(unitFilterLabel.html()).toMatchSnapshot();
});
