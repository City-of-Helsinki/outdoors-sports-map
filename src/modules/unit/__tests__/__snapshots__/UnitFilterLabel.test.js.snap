// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `"<div class=\\"unit-filter-label\\"><span>sport:</span></div>"`;
